### ISSUE TITLE
False positive on SC2259 with cat <<EOF

### ISSUE THREAD

**Issue by pickfire** [koalaman/shellcheck#2369]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2259
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

I want to cat but to a place where only root can access without changing uid for the whole script and without printing it out.

```sh
#!/bin/sh
cat | sudo tee /tmp/test >/dev/null <<EOF
hello
world
EOF
```

#### Here's what shellcheck currently says:

```

In test.sh line 2:
cat | sudo tee /tmp/test >/dev/null <<EOF
                                      ^-^ SC2259 (error): This redirection overrides piped input. To use both, merge or pass filenames.

For more information:
  https://www.shellcheck.net/wiki/SC2259 -- This redirection overrides piped ...
```

#### Here's what I wanted or expected to see:

Not sure, maybe no false positive? At least I search through the wiki and issues but still don't know what to do with this.

**Comment by xabru** [comment]
```shell
 #!/bin/sh
 cat <<EOF | sudo tee /tmp/test >/dev/null
 hello
 world
 EOF
 ```

**Comment by pickfire** [comment]
Thanks, I didn't know that works.

**Comment by koalaman** [comment]
Note that the `cat` is redundant here. You can just do:

```
#!/bin/sh
sudo tee /tmp/test >/dev/null << EOF
hello
world
EOF
```
